.github/workflows/decision.yml

```diff
@@ -0,0 +1,80 @@
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+    - name: Authorization
+      run: |
+        # Get username of GitHub account that triggered this flow
+        opener=${{ github.actor }}
+        
+        # Get list of approved contributors
+        contributors=$(curl https://api.github.com/repos/Monstarrrr/Rebutify/contributors | jq -r '[.[] | select(.type == "User") | .login] | join(",")')
+        
+        # Convert the comma-separated string to an array
+        IFS=',' read -r -a contributor_array <<< "$contributors"
+
+        # Check if triggering user is in the array
+        if [[ ! " ${contributor_array[@]} " =~ " $opener " ]]; then
+          echo "Triggering user is unauthorized."
+          exit 1
+        fi
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Extract subject and category from issue title
+      id: extract_info
+      run: |
+        title="${{ github.event.issue.title }}"
+        if [[ "$title" =~ ^Opening\ decision\ on\ ([^/]+)(/([^/]+))?$ ]]; then
+          subject=$(echo "${BASH_REMATCH[1]}" | tr -d ' ')
+          category=$(echo "${BASH_REMATCH[3]}" | tr -d ' ')
+          if [[ -z "$category" ]]; then
+            category=""
+          fi
+          echo "::set-output name=subject::$subject"
+          echo "::set-output name=category::$category"
+        else
+          echo "Issue title does not follow the required format: 'Opening decision on [category]/[subject]'"
+          exit 1
+        fi
+    - name: Create new branch
+      run: |
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub Action"
+        git checkout -b "docs/${{ github.event.issue.number }}_decision-${{ steps.extract_info.outputs.fileName }}"
+
+        # Determine the directory structure
+        if [[ -z "${{ steps.extract_info.outputs.category }}" ]]; then
+          dir_path="docs/decisions"
+        else
+          dir_path="docs/decisions/${{ steps.extract_info.outputs.category }}"
+        fi
+
+        # Create the directory structure for the documentation
+        mkdir -p "$dir_path"
+        
+        # Create a new markdown file with the issue's body as content
+        echo "${{ github.event.issue.body }}" > "$dir_path/${{ steps.extract_info.outputs.subject }}_DECISION.md"
+        git add .
+        git commit -m "${{ github.event.issue.title }}"
+        git push origin "docs/${{ github.event.issue.number }}_decision-${{ steps.extract_info.outputs.subject }}"
+
+    - name: Create Pull Request
+      uses: peter-evans/create-pull-request@v4
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        commit-message: "Add documentation for decision: ${{ github.event.issue.title }}"
+        title: "Automated PR for issue ${{ github.event.issue.number }}"
+        labels: "automerge"
+        body: "This PR was automatically created to document the issue."
```

Comment:
We could refer the issue in the body instead of the title "was automatically created to document #{{ issue number }}